rllib/examples/checkpoints/continue_training_from_checkpoint.py

```diff
@@ -0,0 +1,213 @@
+                "Reached trigger return of "
+                f"{result[ENV_RUNNER_RESULTS][EPISODE_RETURN_MEAN]}"
+            )
+            self._should_crash = True
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    register_env(
+        "ma_cart", lambda cfg: MultiAgentCartPole({"num_agents": args.num_agents})
+    )
+
+    # Simple generic config.
+    config = (
+        get_trainable_cls(args.algo)
+        .get_default_config()
+        .api_stack(
+            enable_rl_module_and_learner=args.enable_new_api_stack,
+            enable_env_runner_and_connector_v2=args.enable_new_api_stack,
+        )
+        .environment("CartPole-v1" if args.num_agents == 0 else "ma_cart")
+        .env_runners(create_env_on_local_worker=True)
+        .training(lr=0.0001)
+        .callbacks(CrashAfterNIters)
+    )
+
+    # Tune config.
+    # Force-set `args.checkpoint_freq` to 1 and `args.checkpoint_at_end` to True.
+    args.checkpoint_freq = 1
+    args.checkpoint_at_end = True
+    # Need WandB callbacks?
+    tune_callbacks = []
+    if args.wandb_key:
+        project = args.wandb_project or (
+            args.algo.lower() + "-" + re.sub("\\W+", "-", str(config.env).lower())
+        )
+        tune_callbacks.append(
+            WandbLoggerCallback(
+                api_key=args.wandb_key,
+                project=args.wandb_project,
+                upload_checkpoints=False,
+                **({"name": args.wandb_run_name} if args.wandb_run_name else {}),
+            )
+        )
+
+    # Setup multi-agent, if required.
+    if args.num_agents > 0:
+        config.multi_agent(
+            policies={
+                f"p{aid}": PolicySpec(
+                    config=AlgorithmConfig.overrides(
+                        lr=5e-5
+                        * (aid + 1),  # agent 1 has double the learning rate as 0.
+                    )
+                )
+                for aid in range(args.num_agents)
+            },
+            policy_mapping_fn=lambda aid, *a, **kw: f"p{aid}",
+        )
+
+    # Define some stopping criterion. Note that this criterion is an avg episode return
+    # to be reached. The stop criterion does not consider the built-in crash we are
+    # triggering through our callback.
+    stop = {
+        f"{ENV_RUNNER_RESULTS}/{EPISODE_RETURN_MEAN}": args.stop_reward,
+    }
+
+    # Run tune for some iterations and generate checkpoints.
+    tuner = tune.Tuner(
+        trainable=config.algo_class,
+        param_space=config,
+        run_config=air.RunConfig(
+            callbacks=tune_callbacks,
+            checkpoint_config=air.CheckpointConfig(
+                checkpoint_frequency=args.checkpoint_freq,
+                checkpoint_at_end=args.checkpoint_at_end,
+            ),
+            stop=stop,
+        ),
+    )
+    results = tuner.fit()
+    experiment_name = Path(results.experiment_path).name
+
+    # Extract the latest checkpoint from the results and confirm it's the right one.
```

Comment:
Let's state this comment differently. The `get_best_result` gets us only in this specific setup and only with a checkpoint frequency of 1 the latest checkpoint, otherwise we get the one with the highest `episode_return_mean` from whenever this happened.

Reply:
Ah, good catch! Yes, in this example, we should probably just use the last checkpoint, not necessarily the best. ...